src/_utils/focusVisible/index.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { PropsWithChildren, createContext, useEffect, useState } from 'react'
+
+import {
+  KEYS_TRIGGERING_KEYBOARD_NAVIGATION,
+  KEYS_TRIGGERING_KEYBOARD_NAVIGATION_COMPLEMENTARY,
+} from '_utils/keycodes'
+
+// A React hook based on: https://github.com/WICG/focus-visible
+
+const pointerMoveEventList = [
+  'mousedown',
+  'mouseup',
+  'pointermove',
+  'pointerdown',
+  'pointerup',
+  'touchmove',
+  'touchstart',
+  'touchend',
+]
+
+const pointerDownEventList = ['mousedown', 'pointerdown', 'touchstart']
+
+export const FocusVisibleContext = createContext(false)
+
+export const FocusVisibleProvider = (props: PropsWithChildren<{}>) => {
```

Comment:
@simonrelet I use `PropsWithChildren ` as type for children instead of interface like:
```
interface FocusVisibleProviderProps {
  children: ReactNode
}
```
Not typescript expert so just want to know if it's the right way.

Reply:
I have rarely seen `PropsWithChildren` before, but I don't feel comfortable having a component with props but no exported prop types.

If we really want to avoid the `children?: React.ReactNode` part maybe we can write:

```ts
export type FocusVisibleProviderProps = PropsWithChildren<{}>
```

But the empty type (`<{}>`) still feels weird.

Having the extended type declaration (`children?: React.ReactNode`) wouldn't bother me much.

Not sure my comment helps here 😅.

Reply:
I don't really have strong opinion for this point so I'll fallback with:
```
type FocusVisibleProviderProps = {
  children: ReactNode
}
```